src/Compilers/Core/Portable/SourceGeneration/GeneratorState.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+
+#nullable enable
+namespace Microsoft.CodeAnalysis
+{
+    internal readonly struct GeneratorState
```

Comment:
Should this be value-equals?

Reply:
We don't currently compare them anywhere, we put them in a dictionary key'd off the generator. What's the policy in rolsyn, do we usually override equal/gethashcode always, or just when we expect to compare?

---
In reply to: [392521946](https://github.com/dotnet/roslyn/pull/42359#discussion_r392521946) [](ancestors = 392521946)

Reply:
Not necessarily -- if we don't compare them then no need. I thought that we were going to compare them to do up-to-date checks. If that's not the plan, disregard.